source/acquaman/IDEAS/IDEAS2DScanActionController.cpp

```diff
@@ -106,14 +106,7 @@ void IDEAS2DScanActionController::buildScanControllerImplementation()
     transmissionAB->setNormalizationName("Sample");
     scan_->addAnalyzedDataSource(transmissionAB);
 
-    AMXRFDetector *detector = 0;
-
-	if (configuration_->fluorescenceDetector().testFlag(IDEAS::Ketek))
-		detector = qobject_cast<AMXRFDetector *>(AMBeamline::bl()->exposedDetectorByName("KETEK"));
-
-	else if (configuration_->fluorescenceDetector().testFlag(IDEAS::Ge13Element) && IDEASBeamline::ideas()->ge13Element()->isConnected())
-		detector = qobject_cast<AMXRFDetector *>(AMBeamline::bl()->exposedDetectorByName("13-el Ge"));
-
+	AMXRFDetector *detector = IDEASBeamline::ideas()->xrfDetector(configuration_->fluorescenceDetector());
```

Comment:
This only works while the Ketek and the 13 element are never used at the same time.  Since, practically, this is the case we are okay, but I wanted to bring it to your attention.